### ISSUE TITLE
Error: Unable to resolve module storybook/internal/preview-errors

### ISSUE THREAD

**Issue by eduardo-santos-tribia** [storybookjs/react-native#647]
**Describe the bug**
I'm having the error below when opening my app on iOS and Android:

```
Error: Unable to resolve module storybook/internal/preview-errors from /my-project/node_modules/@storybook/addon-actions/dist/preview.mjs: storybook/internal/preview-errors could not be found within the project or in these directories:
  node_modules/@storybook/addon-actions/node_modules
  node_modules
  1 | import { addons } from 'storybook/internal/preview-api';
> 2 | import { ImplicitActionsDuringRendering } from 'storybook/internal/preview-errors';
    |                                                 ^
  3 | import { global } from '@storybook/global';
  4 | import { v4 } from 'uuid';
```

**To Reproduce**
1. run react-native start
2. run react-native run-ios 
3. Metro starts to run and load the app
4. See error when metro loads 100%

**Expected behavior**
The app opens without any storybook-related errors.

**Screenshots**
![Image](https://github.com/user-attachments/assets/c45fd3f3-e498-4dee-bd07-91b9896764ab)

**Code snippets**
metro.config.js
```
const { getDefaultConfig, mergeConfig } = require('@react-native/metro-config')

const path = require('path')

const defaultConfig = getDefaultConfig(__dirname)
const { assetExts, sourceExts } = defaultConfig.resolver

/**
 * Metro configuration
 * https://reactnative.dev/docs/metro
 *
 * @type {import('metro-config').MetroConfig}
 */
const config = {
  transformer: {
    experimentalImportSupport: false,
    inlineRequires: true,
    babelTransformerPath: require.resolve(
      'react-native-svg-transformer/react-native',
    ),
  },
  resolver: {
    assetExts: assetExts.filter(ext => ext !== 'svg'),
    sourceExts: [...sourceExts, 'svg', 'mjs'],
  },
}

const mergedConfig = mergeConfig(defaultConfig, config)

// Storybook v8
const withStorybook = require('@storybook/react-native/metro/withStorybook')
const storybookOptions = {
  // Set to false to remove storybook specific options
  // you can also use a env variable to set this
  enabled: true,

  // set this to 'true' to remove storybook from the bundle when "enabled" is set to 'false'
  onDisabledRemoveStorybook: true,

  // Path to your storybook config
  configPath: path.resolve(__dirname, './.storybook'),
}

module.exports = withStorybook(mergedConfig, storybookOptions)

```

storybook.requires.ts
```
/* do not change this file, it is auto generated by storybook. */

import { start, updateView } from "@storybook/react-native";

import "@storybook/addon-ondevice-actions/register";
import "@storybook/addon-ondevice-notes/register";

const normalizedStories = [
  {
    titlePrefix: "",
    directory: "./src",
    files: "**/*.stories.tsx",
    importPathMatcher:
      /^\.(?:(?:^|\/|(?:(?:(?!(?:^|\/)\.).)*?)\/)(?!\.)(?=.)[^/]*?\.stories\.tsx)$/,
    // @ts-ignore
    req: require.context(
      "../src",
      true,
      /^\.(?:(?:^|\/|(?:(?:(?!(?:^|\/)\.).)*?)\/)(?!\.)(?=.)[^/]*?\.stories\.tsx)$/
    ),
  },
];

declare global {
  var view: ReturnType<typeof start>;
  var STORIES: typeof normalizedStories;
}

const annotations = [
  require("./preview"),
  require("@storybook/react-native/dist/preview"),
  require("@storybook/addon-actions/preview"),
];

global.STORIES = normalizedStories;

// @ts-ignore
module?.hot?.accept?.();

if (!global.view) {
  global.view = start({
    annotations,
    storyEntries: normalizedStories,
  });
} else {
  updateView(global.view, annotations, normalizedStories);
}

export const view = global.view;
```

**System:**
MacOS Sequoia 15.1
React native v0.76.1 (new arch disabled)
"@storybook/addon-ondevice-actions": "^8.3.10",
"@storybook/addon-ondevice-notes": "^8.3.10",
"@storybook/react-native": "^8.3.10",

**Comment by dannyhw** [comment]
do you have the `storybook` package installed? I believe its included with storybook/react-native but maybe theres some issue with dependency resolution.

Can you share a reproduction of the problem?

**Comment by eduardo-santos-tribia** [comment]
No, these are the storybook packages I have installed: 
```
"@storybook/addon-ondevice-actions": "^8.3.10"
"@storybook/addon-ondevice-notes": "^8.3.10"
"@storybook/react-native": "^8.3.10"
```

Even disabling the storybook in metro.config I'm still having this issue. Unfortunately, I can't share the branch and the repo since it's private in my company.

**I forgot to mention that it was working fine in react-native v0.75.4.**

**Comment by dannyhw** [comment]
When i say a reproduction that means setup a new public repository where you recreate the problem so i can investigate

**Comment by eduardo-santos-tribia** [comment]
I'll try to provide a public repository, meanwhile, do you know why my storybook.requires.ts is changing the annotations part?

v.0.75.4
```
const annotations = [
  require("./preview"),
  require("@storybook/react-native/dist/preview"),
  require("@storybook/addon-actions/preview"),
];
```

v0.76.1
```
const annotations = [
  require("./preview"),
  require("@storybook/react-native/preview"),
  require("@storybook/addon-ondevice-actions/preview"),
];
```

**Comment by dannyhw** [comment]
This is from storybook 8.4 its been updated

Requires is generated so you don't need to manually edit it

You dont have a pinned version so looks like you updated your storybook version too

**Comment by eduardo-santos-tribia** [comment]
@dannyhw adding `"storybook": "^8.4.2"` as `devDependency` solved the issue, but I think would be good if you could check if this is a requirement or is just something related to my environment, since you mentioned that this should be part of `@storybook/react-native`. It might be related to the new `@storybook/react-native v8.4.+`.

**Comment by dannyhw** [comment]
@eduardo-santos-tribia this is already a dependency of storybook/react-native like you say, you can see it here

https://github.com/storybookjs/react-native/blob/210de916b72f198a9732397336022072987c7563/packages/react-native/package.json#L63C3-L63C27

The problem is probably that not every package you had got updated to 8.4 the fix is just to update everything to 8.4 and it shouldn't resolve the wrong version anymore I imagine

**Comment by eduardo-santos-tribia** [comment]
@dannyhw all my  `package.json` storybook dependencies are in v8.4, should I check something specific in `yarn.lock`? I've checked and all storybook related dependencies are on v8.4.2 in my yarn.lock file.

**Comment by dannyhw** [comment]
maybe you can try a clean install after deleting node_modules etc, alternatively if installing the package fixes things then just stick with that solution for now.

**Comment by eduardo-santos-tribia** [comment]
@dannyhw I tried to clean `node_modules` and didn't work. I'll close this issue and keep the storybook dependency for now. Thanks for the help.

**Comment by danoc** [comment]
Hi, I ran into this issue as well. The problem, in my case, is Yarn is installing `storybook` in `node_modules/@storybook/react-native/node_modules/storybook`, making it impossible for `node_modules/@storybook/addon-actions` to access that dependency. 

Installing `storybook` separately works because it gets Yarn to make it available at `node_modules/storybook`, a path that `node_modules/@storybook/addon-actions` can access.

Yarn's behavior is reasonable, IMO. `@storybook/react-native` depends on `storybook` and Yarn is choosing to install it in `node_modules/@storybook/react-native/node_modules/storybook`.

I'd recommend making `storybook` a peer dependency of `@storybook/react-native` rather than a dependency or updating the documentation to recommend installing `storybook` separately (as we've identified as a workaround in this thread). The downside of the latter approach is that developers would need to be careful to not end up with multiple versions of Storybook within their `node_modules`. This would be an easy mistake to make.
